pvlib/tests/test_albedo.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import pandas as pd
+import pytest
+from pvlib import albedo
+
+from .conftest import assert_series_equal
+from numpy.testing import assert_allclose
+
+
+def test_albedo_water_default():
+    result = albedo.inland_water_dvoracek(solar_elevation=90,
+                                          color_coeff=0.13,
+                                          wave_roughness_coeff=0.29)
+    assert_allclose(result, 0.072, 0.001)
+
+
+def test_albedo_water_negative_elevation():
+    result = albedo.inland_water_dvoracek(solar_elevation=-60,
+                                          color_coeff=0.13,
+                                          wave_roughness_coeff=0.29)
+    assert_allclose(result, 0.13, 0.01)
+
+
+def test_albedo_water_string_surface_condition():
+    result = albedo.inland_water_dvoracek(solar_elevation=90,
+                                          surface_condition='clear_water_no_waves')   # noqa: E501
+    assert_allclose(result, 0.072, 0.001)
+
+
+def test_albedo_water_ndarray():
+    solar_elevs = np.array([-50, 0, 20, 60, 90])
+    color_coeffs = np.array([0.1, 0.1, 0.2, 0.3, 0.4])
+    roughness_coeffs = np.array([0.3, 0.3, 0.8, 1.5, 2])
+    result = albedo.inland_water_dvoracek(solar_elevation=solar_elevs,
+                                          color_coeff=color_coeffs,
+                                          wave_roughness_coeff=roughness_coeffs)   # noqa: E501
+    expected = np.array([0.1, 0.1, 0.12875, 0.06278, 0.064])
+    assert_allclose(expected, result, atol=1e-5)
+
+
+def test_albedo_water_series():
+    times = pd.date_range(start="2015-01-01 00:00", end="2015-01-02 00:00",
+                          freq="6h")
+    solar_elevs = pd.Series([-50, 0, 20, 60, 90], index=times)
+    color_coeffs = pd.Series([0.1, 0.1, 0.2, 0.3, 0.4], index=times)
+    roughness_coeffs = pd.Series([0.1, 0.3, 0.8, 1.5, 2], index=times)
+    result = albedo.inland_water_dvoracek(solar_elevation=solar_elevs,
+                                          color_coeff=color_coeffs,
+                                          wave_roughness_coeff=roughness_coeffs)   # noqa: E501
+    expected = pd.Series([0.1, 0.1, 0.12875, 0.06278, 0.064], index=times)
+    assert_series_equal(expected, result, atol=1e-5)
+
+
+def test_albedo_water_invalid():
+    with pytest.raises(ValueError):
+        albedo.inland_water_dvoracek(45)
```

Comment:
I'd probably make this test a bit more explicit. E.g., test that a ValueError is raised when passing 'not_a_surface_condition' to `surface_conditions`.

Also, test that a ValueError is raised when only the `color_coeff` is specified and `surface_roughness_coeff` is not. These tests can be under the same function (test_albedo_water_invalid) but with more `with pytests..` statements after each other.